src/pages/index.js

```diff
@@ -68,3 +70,15 @@ export default function Home() {
     </div>
   );
 }
+
+//Initializing mongo connection
+export async function getServerSideProps() {
+  await connectMongo();
```

Comment:
to further clarify-- ideally we should **never** have to call this function during "runtime". it should be called when the sever is first starting, by placing it on the top level of a file to be run when it's imported (ie the Test.js file-- or even the schema file). LMK if this doesn't work though-- i'm not super familiar with how mongodb handles connections.

Reply:
I'm not sure if it's what wanted, but I can make a top-level await at service action files.

Reply:
I have spent days on this top-level-await feature, but I can't identify the problem, the database is connected but the query still timeout, hence I've arrived with the current implementation:
1. in index.js make a service side prompt and check the status of `global.mongoose.db` - this variable stores our cached db connection
2. if `global.mongoose`.db does not exist, call `await connectMongo()` to establish the connection
Ideally for any other client-side file with exception of index.js, we don't need above process

Reply:
My idea for future work flow:
1. In the index page we initialize the mongodb connection by call `await connectMongo()` as I did in current implementation, we ignore the variable returned and render our page as if this function do not exist
2. in any other page we may fetch data without call `connectMongo()` at all(we don't call `connectMongo()`  in service side action such as test.js)

I personally believe that this workflow does meet the expectation.